src/Eris/Generator/SuchThatGenerator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+namespace Eris\Generator;
+
+use Eris\Generator;
+use LogicException;
+
+/**
+ * TODO: maybe filter?
+ */
+function suchThat(callable $filter, Generator $generator)
+{
+    return new SuchThatGenerator($filter, $generator);
+}
+
+class SuchThatGenerator implements Generator
+{
+    private $filter;
+    private $generator;
+    private $maximumAttempts;
+    
+    public function __construct(callable $filter, $generator)
+    {
+        $this->filter = $filter;
+        $this->generator = $generator;
+        $this->maximumAttempts = 10;
+    }
+
+    public function __invoke($size)
+    {
+        $value = $this->generator->__invoke($size);
+        $attempts = 0;
+        while (!$this->predicate($value)) {
+            if ($attempts >= $this->maximumAttempts) {
+                throw new LogicException("Tried to satisfy predicate $attempts times, but could not generate a good value. Last one was: " . $value);
```

Comment:
Can we provide a better error message here? We can keep this one but maybe we can add something useful to solve the problem, something like "You should try to improve your generator".